File: scripts/create_simulator_user.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');

// Manual .env parsing
const envPath = path.join(__dirname, '../server/.env');
if (fs.existsSync(envPath)) {
  fs.readFileSync(envPath, 'utf-8').split('\n').forEach(line => {
    const [key, ...valueParts] = line.split('=');
    const value = valueParts.join('=').trim().replace(/^["']|["']$/g, '');
    if (key && key.trim()) process.env[key.trim()] = value;
  });
}

const { Pool } = require('pg');

const pool = new Pool({
  connectionString: process.env.DB_URL || `postgresql://${process.env.DB_USER || 'postgres'}:${process.env.DB_PASSWORD || 'postgres'}@${process.env.DB_HOST || 'localhost'}:${process.env.DB_PORT || 5432}/${process.env.DB_NAME || 'api_traffic'}`,
});

(async () => {
  const client = await pool.connect();
  try {
    const res = await client.query(
      'INSERT INTO users (id, email, password_hash, created_at) VALUES ($1, $2, $3, NOW()) ON CONFLICT (id) DO NOTHING RETURNING id, email;',
      [999, 'simulator@traffic-intel.local', 'dummy_hash']
    );
    if (res.rows[0]) {
      console.log('✅ Created user 999:', res.rows[0]);
    } else {
      console.log('ℹ️ User 999 already exists');
    }
  } catch (e) {
    console.error('❌ Error:', e.message);
    process.exit(1);
  } finally {
    client.release();
    await pool.end();
  }
})();

File: scripts/create_simulator_user.py
#!/usr/bin/env python3
import os
import psycopg2

# Parse .env file
env_file = 'server/.env'
for line in open(env_file):
    if '=' in line and not line.startswith('#'):
        k, v = line.strip().split('=', 1)
        os.environ[k.strip()] = v.strip().strip('"').strip("'")

# Connect and create user
conn_string = os.getenv('DB_URL') or f"postgresql://{os.getenv('DB_USER', 'postgres')}:{os.getenv('DB_PASSWORD', 'postgres')}@{os.getenv('DB_HOST', 'localhost')}:{os.getenv('DB_PORT', '5432')}/{os.getenv('DB_NAME', 'api_traffic')}"

try:
    conn = psycopg2.connect(conn_string)
    cur = conn.cursor()
    cur.execute("""
        INSERT INTO users (id, email, password)
        VALUES (%s, %s, %s)
        ON CONFLICT (id) DO UPDATE SET email=EXCLUDED.email
        RETURNING id, email;
    """, [999, 'simulator@traffic-intel.local', 'dummy_password'])
    result = cur.fetchone()
    conn.commit()
    print(f'✅ Created/updated user {result[0]}: {result[1]}')
    conn.close()
except Exception as e:
    print(f'❌ Error: {e}')
    exit(1)
